### PR TITLE
Changed

### LINKED ISSUE #215: Delete All User Data On Logout
It Consists of clearing shared pref and and Database

### PR DESCRIPTION
**Pull request by SaumyaSingh1**
Fixes #215

### CODE REVIEW

**Comment by aggarwalpulkit596** [comment]
Clean database as well that's is the main user data as well user downloaded video files

**Comment by SaumyaSingh1** [comment]
okay

File: app/src/main/java/com/codingblocks/cbonlineapp/PreferenceManager.kt [hunk 1]
```diff
@@ -0,0 +1,35 @@
+package com.codingblocks.cbonlineapp
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.content.SharedPreferences
+
+class PreferenceManager {
```
Comment (adityastic) [inline app/src/main/java/com/codingblocks/cbonlineapp/PreferenceManager.kt]: can you explain why you created a new class?

File: app/src/main/java/com/codingblocks/cbonlineapp/activities/HomeActivity.kt [hunk 2]
```diff
@@ -146,6 +151,21 @@ class HomeActivity : AppCompatActivity(), NavigationView.OnNavigationItemSelecte
             createShortcut()
         }
         nav_view.getHeaderView(0).login_button.setOnClickListener {
+            val builder = AlertDialog.Builder(this)
```
Comment (adityastic) [inline app/src/main/java/com/codingblocks/cbonlineapp/activities/HomeActivity.kt]: why are you coding inside the login section. You have to delete everything on logout, and not when everytime a user clicks on the login button. check the specific condition of logout and code in that please

File: app/src/main/java/com/codingblocks/cbonlineapp/PreferenceManager.kt (hunk 1 above)
Comment (SaumyaSingh1) [reply app/src/main/java/com/codingblocks/cbonlineapp/PreferenceManager.kt]: Although could have done  using PreferenceHelper , but for having clarity about clearing sharedprefs I created new class PreferenceManager

Comment (adityastic) [reply app/src/main/java/com/codingblocks/cbonlineapp/PreferenceManager.kt]: @aggarwalpulkit596

Comment (adityastic) [reply app/src/main/java/com/codingblocks/cbonlineapp/PreferenceManager.kt]: im not sure if you should clear a seperate class for handling preferences. all you need is the clear prefs. all the rest is useless, you should copy the function in the utils and use it

File: app/src/main/java/com/codingblocks/cbonlineapp/activities/HomeActivity.kt (hunk 2 above)
Comment (SaumyaSingh1) [reply app/src/main/java/com/codingblocks/cbonlineapp/activities/HomeActivity.kt]: ` nav_view.getHeaderView(0).login_button.text = "Logout"`
clicklistener is applied only after text of login-button is LOGOUT.

Comment (adityastic) [reply app/src/main/java/com/codingblocks/cbonlineapp/activities/HomeActivity.kt]: arnt you coding directly in the login clicklistener? rather coding in the condition specific to logout?

Comment (adityastic) [reply app/src/main/java/com/codingblocks/cbonlineapp/activities/HomeActivity.kt]: please check the code, we are removing shortcuts and invalidating tokens on logout............ code in that block, this code is prone to fail

**Comment by SaumyaSingh1** [comment]
@aggarwalpulkit596  Have cleared the logout shared prefs and database.

File: app/src/main/java/com/codingblocks/cbonlineapp/activities/HomeActivity.kt (hunk 2 above)
Comment (SaumyaSingh1) [reply app/src/main/java/com/codingblocks/cbonlineapp/activities/HomeActivity.kt]: when the loginbtntext is not logout then we already have a listener that starts an intent .
`  nav_view.getHeaderView(0).login_button.setOnClickListener {
            startActivity(intentFor<LoginActivity>().singleTop())
        }`

Comment (adityastic) [reply app/src/main/java/com/codingblocks/cbonlineapp/activities/HomeActivity.kt]: @SaumyaSingh1 Oh wait, I get your point. The codebase has issues itself. Remove this condition as well 
`if (nav_view.getHeaderView(0).login_button.text == "Logout") `
it is useless

Comment (SaumyaSingh1) [reply app/src/main/java/com/codingblocks/cbonlineapp/activities/HomeActivity.kt]: @adityastic If condition `if (nav_view.getHeaderView(0).login_button.text == "Logout") `  will be removed then how will we reach condition specific to logout ? It's the only condition that takes us to the logout ..

Comment (SaumyaSingh1) [reply app/src/main/java/com/codingblocks/cbonlineapp/activities/HomeActivity.kt]: @aggarwalpulkit596 please check if the code is done in clicklistener of logout or not .

Comment (adityastic) [reply app/src/main/java/com/codingblocks/cbonlineapp/activities/HomeActivity.kt]: We assign the text to logout just before the click listener is set. How does checking it again make sense?. Your code is correct, you can also remove this condition as it is redundant

Comment (SaumyaSingh1) [reply app/src/main/java/com/codingblocks/cbonlineapp/activities/HomeActivity.kt]: @adityastic  Thankyou for your reviews :)

File: app/src/main/java/com/codingblocks/cbonlineapp/PreferenceManager.kt (hunk 1 above)
Comment (aggarwalpulkit596) [reply app/src/main/java/com/codingblocks/cbonlineapp/PreferenceManager.kt]: not needed use preference helper

File: app/src/main/java/com/codingblocks/cbonlineapp/activities/HomeActivity.kt (hunk 2 above)
Comment (aggarwalpulkit596) [reply app/src/main/java/com/codingblocks/cbonlineapp/activities/HomeActivity.kt]: Use the custom dialog for confirmation

Comment (aggarwalpulkit596) [reply app/src/main/java/com/codingblocks/cbonlineapp/activities/HomeActivity.kt]: `if (nav_view.getHeaderView(0).login_button.text == "Logout")` you can remove this condition it's useless

File: app/src/main/java/com/codingblocks/cbonlineapp/activities/HomeActivity.kt [hunk 3]
```diff
@@ -421,4 +441,9 @@ class HomeActivity : AppCompatActivity(), NavigationView.OnNavigationItemSelecte
                 println("Failed to delete database journal")
         }
     }
+
+    override fun onSupportNavigateUp(): Boolean {
```
Comment (aggarwalpulkit596) [inline app/src/main/java/com/codingblocks/cbonlineapp/activities/HomeActivity.kt]: why added this ?

**Comment by SaumyaSingh1** [comment]
Have commited the changes , please check @aggarwalpulkit596

File: app/src/main/java/com/codingblocks/cbonlineapp/activities/HomeActivity.kt [hunk 4]
```diff
@@ -146,12 +147,27 @@ class HomeActivity : AppCompatActivity(), NavigationView.OnNavigationItemSelecte
...
+            val dialog: AlertDialog = builder.create()
+            dialog.show()
             prefs.SP_ACCESS_TOKEN_KEY = PreferenceHelper.ACCESS_TOKEN
             prefs.SP_JWT_TOKEN_KEY = PreferenceHelper.JWT_TOKEN
-            if (nav_view.getHeaderView(0).login_button.text == "Logout") {
                 removeShortcuts()
                 invalidateToken()
-            }
             startActivity(intentFor<LoginActivity>().singleTop())
             finish()
```
Comment (adityastic) [inline app/src/main/java/com/codingblocks/cbonlineapp/activities/HomeActivity.kt]: All the six lines above should execute only when the user confirms the logout.

**Comment by SaumyaSingh1** [comment]
should I make a separate PR with only 1 commit @aggarwalpulkit596  
Because last time also you asked to have one commit per PR.

**Comment by aggarwalpulkit596** [comment]
Firstly resolve conflicts and delete downloaded videos as well

**Comment by SaumyaSingh1** [comment]
I am not able to refer to downloaded videos . And I am resolving conflicts .

**Comment by SaumyaSingh1** [comment]
I am trying to clear the downloaded video part as well.

**Comment by SaumyaSingh1** [comment]
Conflicts resolved @aggrawalpulkit

File: app/src/main/java/com/codingblocks/cbonlineapp/activities/HomeActivity.kt (hunk 4 above)
Comment (adityastic) [reply app/src/main/java/com/codingblocks/cbonlineapp/activities/HomeActivity.kt]: @aggrawalpulkit596

Comment (adityastic) [reply app/src/main/java/com/codingblocks/cbonlineapp/activities/HomeActivity.kt]: @aggrawalpulkit596

Comment (aggarwalpulkit596) [reply app/src/main/java/com/codingblocks/cbonlineapp/activities/HomeActivity.kt]: @SaumyaSingh1 firstly correct the logic and secondly there are merge heads in your code fix that too read about github in the contribution guidelines you'll know

**Pull request closed without merging**